### PR TITLE
[P8-64] Add fischer lcgs

### PR DESCRIPTION
**Pull request by MathiasMehl**
Adds fischer mutex algorithm. Inspiration taken from UPPAAL case study. UPPAAL ships with a fischer implementation, see picture below
image

This has been implemented in a series of lcgs programs with varying number of processes. Formulae, structure in directories/files are similar to peterson in PR #107

### CODE REVIEW

**Comment by NicEastvillage** [comment]
PR #99 has been merged into main. Please merge main into your branch and move the Fischer LCGS to the `lcgs-examples` folder.

File: lcgs-examples/fischer/1/multiple-in-CS-FALSE-01.atl [hunk 1]
```diff
@@ -0,0 +1 @@
+<<p0,p1>> F (multiple_in_cs)
```
Comment (NicEastvillage) [inline lcgs-examples/fischer/1/multiple-in-CS-FALSE-01.atl:1]: Not equivalent to the json version. The number of players varies.

File: lcgs-examples/fischer/1/p1-eventually-reach-CS-TRUE-01.atl [hunk 2]
```diff
@@ -0,0 +1 @@
+<<p0,p1>> F (p0_in_cs)
```
Comment (NicEastvillage) [inline lcgs-examples/fischer/1/p1-eventually-reach-CS-TRUE-01.atl:1]: Not equivalent to the json version. The number of players varies.

File: lcgs-examples/fischer/3/ensure-mutual-exclusion-TRUE-03.atl [hunk 3]
```diff
@@ -0,0 +1 @@
+[[p0,p1,p2]] G (mutual_exclusion)
```
Comment (NicEastvillage) [inline lcgs-examples/fischer/3/ensure-mutual-exclusion-TRUE-03.atl:1]: The LCGS does not have a label called `mutual exclusion`

File: lcgs-examples/fischer/2/ensure-mutual-exclusion-TRUE-02.atl [hunk 4]
```diff
@@ -0,0 +1 @@
+[[p0,p1]] G (mutual_exclusion)
```
Comment (NicEastvillage) [inline lcgs-examples/fischer/2/ensure-mutual-exclusion-TRUE-02.atl:1]: The LCGS does not have a label called `mutual exclusion`

File: lcgs-examples/fischer/10/ensure-mutual-exclusion-TRUE-10.atl [hunk 5]
```diff
@@ -0,0 +1 @@
+[[p0,p1,p2,p3,p4,p5,p6,p7,p8,p9]] G (mutual_exclusion)
```
Comment (NicEastvillage) [inline lcgs-examples/fischer/10/ensure-mutual-exclusion-TRUE-10.atl:1]: The LCGS does not have a label called `mutual exclusion`

File: lcgs-examples/fischer/1/ensure-mutual-exclusion-TRUE-01.atl [hunk 6]
```diff
@@ -0,0 +1 @@
+[[p0,p1]] G (mutual_exclusion)
```
Comment (NicEastvillage) [inline lcgs-examples/fischer/1/ensure-mutual-exclusion-TRUE-01.atl:1]: The LCGS does not have a label called `mutual exclusion`

**Review by NicEastvillage — CHANGES_REQUESTED** [review]

File: lcgs-examples/fischer/1/ensure-mutual-exclusion-TRUE-01.atl (hunk 6 above)
Comment (MathiasMehl) [reply lcgs-examples/fischer/1/ensure-mutual-exclusion-TRUE-01.atl]: I have renamed the label in 4f94bcb

File: lcgs-examples/fischer/1/p1-eventually-reach-CS-TRUE-01.atl (hunk 2 above)
Comment (MathiasMehl) [reply lcgs-examples/fischer/1/p1-eventually-reach-CS-TRUE-01.atl]: Should be fixed in b46fa72

**Review by NicEastvillage — APPROVED** [review]

**Review by lars4509 — APPROVED** [review]

**Pull request merged**